magento/magento2
# 24070 - Allowed multivalue form elements to submit data to website/gr…

## Example

## Issue


### Proposed solution



### Description (*)
Fix for issue: https://github.com/magento/magento2/issues/24070

When submitting a website, group or store entity form (add/edit), submitted multivalue form elements get filtered and only the last selected element will be posted to the controller.

This posted data
```
fieldname[]: value1
fieldname[]: value2
```
Turns into
```
{
   fieldname[]: value2
}
```

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/24070#issue-478488652 : Website/Group/Store entity form can't be extended with multiple value form element for extension attributes

* Transform fieldnames with multiple [] to an indexed one.
** This: 
```
fieldname[]: value1
fieldname[]: value2
```
** Turns into
```
fieldname[]: value1
fieldname[1]: value2
```

Therefore the names are different and both fields can be converted to a new input to be actually posted to the controller.

I chose not to convert the first fieldname[] to fieldname[0] because altering this, I could not reference to the fieldname within the same set of data and this needed more javascript code. This is not necessary for PHP which receives fieldname[] as 0 and fieldname[1] as 1.

Used let/const because they are supported in IE11 and don't interfere with scope of the function.

### Manual testing scenarios (*)
* Extend the Store.php form block with a multivalue field. See: https://github.com/magento/magento2/issues/24070
** <select name="fieldname[]" multiple="multiple">
* The submitted data in validate-store.js will converted a key/value object
* A new html form will be generated with <input name="" value="" /> of the previous object entries.
* This form is posted to the controller.
* Current postdata fields won't be altered, multivalue fields will be submitted, before only the last element gets posted.
